src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -432,18 +432,22 @@ public void Flush()
             this.configuration.TelemetryChannel.Flush();
         }
 
-        private string GetAssemblyVersion()
+        private string GetSdkVersion()
         {
 #if !CORE_PCL
-            return VersionPrefix + typeof(TelemetryClient).Assembly.GetCustomAttributes(false)
+            string versionStr = typeof(TelemetryClient).Assembly.GetCustomAttributes(false)
                     .OfType<AssemblyFileVersionAttribute>()
                     .First()
                     .Version;
+            
 #else
-            return VersionPrefix + typeof(TelemetryClient).GetTypeInfo().Assembly.GetCustomAttributes<AssemblyFileVersionAttribute>()
+            string versionStr = typeof(TelemetryClient).GetTypeInfo().Assembly.GetCustomAttributes<AssemblyFileVersionAttribute>()
                     .First()
                     .Version;
 #endif
+
+            Version version = new Version(versionStr);
+            return VersionPrefix + version.ToString(3) + "-" + version.Revision;
```

Comment:
When we have version 2.1.0.25000 than 25000 is a revision and version.ToString(3) gives 2.1.0. That is verified by unit test.

---
In reply to: [65466084](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/234#discussion_r65466084) [](ancestors = 65466084)